### PR TITLE
[build] set Thread 1.2 as default version

### PR DESCRIPTION
**Pull request by jwhui**
This PR includes a number of minor fixes necessary for GitHub Actions
checks to pass.

### CODE REVIEW

File: .github/workflows/posix.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
       CXXFLAGS: -DCLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER=1 -DOPENTHREAD_CONFIG_MLE_MAX_CHILDREN=15 -DOPENTHREAD_CONFIG_MAC_SCAN_DURATION=500
       CC: clang
       CXX: clang++
+      THREAD_VERSION: 1.1
```
Comment (jwhui) [inline .github/workflows/posix.yml:50]: Thread 1.2 multicast feature does not build on macOS.

File: .github/workflows/posix.yml [hunk 2]
```diff
@@ -142,6 +143,7 @@ jobs:
       PYTHONUNBUFFERED: 1
       READLINE: readline
       REFERENCE_DEVICE: 1
+      THREAD_VERSION: 1.1
```
Comment (jwhui) [inline .github/workflows/posix.yml:146]: Tests do not pass with 1.2 enabled.

File: .github/workflows/posix.yml [hunk 3]
```diff
@@ -182,6 +184,7 @@ jobs:
...
```
Comment (jwhui) [inline .github/workflows/posix.yml:187]: Tests do not pass with 1.2 enabled.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

We can add 1.2 tests later

File: .github/workflows/simulation-1.1.yml [hunk 4]
```diff
@@ -316,6 +323,8 @@ jobs:
 
   external-commissioner:
     runs-on: ubuntu-20.04
+    env:
+      THREAD_VERSION: 1.1
```
Comment (wgtdkp) [inline .github/workflows/simulation-1.1.yml:327]: But the CI tests on `ot-commissioner` master will be broken.

Comment (jwhui) [reply .github/workflows/simulation-1.1.yml:327]: Do you have a suggested path forward?

**Review by abtink — APPROVED** [review]
👍

File: .github/workflows/simulation-1.1.yml (hunk 4 above)
Comment (wgtdkp) [reply .github/workflows/simulation-1.1.yml:327]: I think I misunderstood this change. I thought there are some external commissioner failures that force you setting the `THREAD_VERSION` to 1.1 for this test case, but seems we are just using 1.1 for all test cases in this workflow and the workflow itself is also renamed to `simulation-1.1.yml`.

Actually, the external-commissioner test script doesn't respect the `THREAD_VERSION` env variable and it actually tests with latest openthread master commit and the CI on `ot-commissioner` master  will conitnue working.

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Comment by bukepo** [comment]
Do we still need check size diff for 1.1?

**Comment by jwhui** [comment]
> Do we still need check size diff for 1.1?

Good question. How about we address this in a separate PR?

**Comment by bukepo** [comment]
> > Do we still need check size diff for 1.1?
> 
> Good question. How about we address this in a separate PR?

Sure!

**Pull request merged**